Client/Ecosia/UI/NTP/DefaultBrowser.swift

```diff
@@ -22,6 +22,21 @@ final class DefaultBrowser: UIViewController, NotificationThemeable {
     weak var cta: UIButton!
     weak var skip: UIButton!
     weak var delegate: DefaultBrowserDelegate?
+    
+    var text1String: String {
+        if IncentivisedSearchChecker.isSearchIncentiveRestricted {
```

Comment:
I wonder if it would be smarter (and more visible) to have a central class/extension for those checks 🤔 Maybe even inside the `String.localised` logic. Otherwise we'll have these `if else` or ternary expression scattered through the codebase which is also ok, but not really great imho.

Reply:
My first thought was also having a helper method like `String.localized(_ forKey: Key, withSearchIncentiveRestrictedAlternativeKey alternativeKey: Key)`, but I realised most cases are not only text and require some additional logic, e.g. onboarding, your impact. 

Even so, we have already mapped at least two places were it is only text (this one and referrals), so I believe you are right that this would be smarter and less code as re-usable logic, so will change back to that 🙂